### PR TITLE
Rename mocha tests to use *.test extension

### PR DESCRIPTION
**Pull request by eessex**
Related to work installing codecov...
Our current method for finding mocha tests is not comprehensive because it looks for a nested test directory. This means that we are missing tests when directories are named `tests` rather than `test` or if they do not match the existing find regex `$(find src/desktop/apps/*/**/*/test -name '*.coffee')`.

This PR does a find-replace on mocha test files to rename extensions as `*.test.coffee` or `*.test.js`, which allows us to search for an extension and eliminates the need for diving directories in our current setup. It also catches a number of tests that were not previously run.

Renames tests in directories:
- `src/desktop/components`
- `src/desktop/test`
- `src/test`

### CODE REVIEW

File: src/desktop/components/branching_state/test/index.test.coffee [hunk 1]
```diff
@@ -9,7 +9,7 @@ describe 'State', ->
       state.context.state.should.equal state
 
     describe '#inject', ->
-      it 'lets you inject additions to the context at any point', ->
+      xit 'lets you inject additions to the context at any point', ->
```
Comment (damassi) [inline src/desktop/components/branching_state/test/index.test.coffee]: Assuming this is deliberate?

File: scripts/test.sh [hunk 2]
```diff
@@ -7,13 +7,9 @@ set -ex
...
-    yarn mocha src/test/lib/*
-    yarn mocha $(find src/desktop/test -name '*.coffee')
-    yarn mocha $(find src/desktop/components/*/test -name '*.coffee')
-    yarn mocha $(find src/desktop/components/*/test -name '*.js')
-    yarn mocha $(find src/desktop/components/**/*/test -name '*.coffee')
-    yarn mocha $(find src/desktop/components/**/*/test -name '*.js')
-    yarn mocha $(find src/desktop/components -name '*.test.js')
+    yarn mocha $(find src/test -name '*.test.*')
+    yarn mocha $(find src/desktop/test -name '*.test.*')
+    yarn mocha $(find src/desktop/components -name '*.test.*')
```
Comment (damassi) [inline scripts/test.sh]: 👌

**Comment by damassi** [comment]
Ah, I was thinking that the code coverage % seemed low for older code -- all of that seems to be pretty well tested.

File: src/desktop/components/branching_state/test/index.test.coffee (hunk 1 above)
Comment (eessex) [reply src/desktop/components/branching_state/test/index.test.coffee]: Yeah this one was broken -- maybe wasn't running before?

**Comment by eessex** [comment]
Will follow up with the other folders tomorrow...

**Pull request merged**